src/20240502_AaveV1Ethereum_AaveV1Deprecation/AaveV1Ethereum_AaveV1Deprecation_20240502.t.sol

```diff
@@ -0,0 +1,293 @@
+          users[i].debt,
+          users[i].user,
+          type(uint256).max,
+          false
+        );
+      } else {
+        deal(users[i].debt, address(this), currentBorrowBalance);
+        assertEq(currentBorrowBalance, IERC20(users[i].debt).balanceOf(address(this)));
+        IERC20(users[i].debt).approve(proposal.ADDRESSES_PROVIDER().getLendingPoolCore(), 0);
+        IERC20(users[i].debt).approve(
+          proposal.ADDRESSES_PROVIDER().getLendingPoolCore(),
+          type(uint256).max
+        );
+        ILendingPool(AaveV1.POOL).liquidationCall(
+          users[i].collateral,
+          users[i].debt,
+          users[i].user,
+          type(uint256).max,
+          false
+        );
+      }
+      (, uint256 totalCollateralETHAfter, uint256 totalBorrowsETHAfter, , , , , ) = ILendingPool(
+        AaveV1.POOL
+      ).getUserAccountData(users[i].user);
+
+      uint256 collateralDiff = totalCollateralETHBefore - totalCollateralETHAfter;
+      uint256 borrowsDiff = totalBorrowsETHBefore - totalBorrowsETHAfter;
+      assertGt(collateralDiff, borrowsDiff);
+      assertApproxEqAbs((borrowsDiff * 1 ether) / collateralDiff, 0.95 ether, 0.006 ether); // should be ~5% + rounding
+    }
+  }
+
+  /**
+   * Users should be able to repay their debt
+   */
+  function test_repay() public {
+    executePayload(vm, address(proposal));
+
+    V1User[] memory users = _getUsers();
+    for (uint256 i = 0; i < users.length; i++) {
+      vm.startPrank(users[i].user);
+
+      // repay
+      _repayV1(users[i].debt, users[i].user);
+
+      vm.stopPrank();
+    }
+  }
+
+  /**
+   * Users should be able to redeem their collateral
+   */
+  function test_redeem() public {
+    executePayload(vm, address(proposal));
+
+    V1User[] memory users = _getUsers();
+
+    for (uint256 i = 0; i < users.length; i++) {
+      vm.startPrank(users[i].user);
+      _repayV1(users[i].debt, users[i].user);
+
+      // redeem
+      _redeemV1(users[i].collateral, users[i].user, 100);
+
+      vm.stopPrank();
+    }
+  }
+
+  /**
+   * Rates should be updated and rate should be zero
+   */
+  function test_rates() public {
+    address[] memory reserves = proposal.CORE().getReserves();
+    uint256 snapshot = vm.snapshot();
+    for (uint256 i = 0; i < reserves.length; i++) {
+      uint256 lIndexBefore = proposal.CORE().getReserveLiquidityCumulativeIndex(reserves[i]);
+      uint256 vIndexBefore = proposal.CORE().getReserveVariableBorrowsCumulativeIndex(reserves[i]);
+      executePayload(vm, address(proposal));
+      uint256 lIndexAfter = proposal.CORE().getReserveLiquidityCumulativeIndex(reserves[i]);
+      uint256 vIndexAfter = proposal.CORE().getReserveVariableBorrowsCumulativeIndex(reserves[i]);
+      uint256 liquidityRateAfter = proposal.CORE().getReserveCurrentLiquidityRate(reserves[i]);
+
+      assertEq(lIndexBefore, lIndexAfter);
+      assertEq(vIndexBefore, vIndexAfter);
+      assertEq(liquidityRateAfter, 0);
```

Comment:
please add to this test, that currentStableBorrowRate and currentVariableBorrowRate also 0